packages/grafana-data/src/types/suggestions.ts

```diff
@@ -0,0 +1,8 @@
+import { DataFrame, PanelPlugin, VariableModel, VariableSuggestion, VariableSuggestionsScope } from '..';
+
+export type SuggestionsProvider = (
+  plugin: PanelPlugin,
+  data: DataFrame[],
+  getTemplateVariables: () => VariableModel[],
```

Comment:
hymm -- not wild about adding `getTemplateVariables` or templateSrv  -- maybe better if the response returns a more complex object like:

```
{
  includeTemplateVariables?: boolean;
  suggestions?: VariableSuggestion[];
}
```

Reply:
actually maybe `includeTemplateVariables` is a flag next to this since it likely does not depend on the state